docs/chains/doc_gen/mdx_adapter.py

```diff
@@ -24,62 +28,182 @@
 """
 
 
-class MDXAdapterTranslator(MarkdownTranslator): ...
+class MDXAdapterTranslator(MarkdownTranslator):
+    pass
 
 
-def extract_and_format_parameters_section(content: str) -> str:
-    def format_as_table(items: list[tuple[str, str, str]]) -> str:
-        header = "| Name | Type | Description |\n|------|------|-------------|\n"
-        rows = [
-            f"| {name} | {typ} | {description} |" for name, typ, description in items
-        ]
-        return header + "\n".join(rows)
-
-    pattern = r"(\* \*\*Parameters:\*\*\n((?: {2}\* .+(?:\n {4}.+)*\n?)+))"
-    matches = re.findall(pattern, content)
-
-    for match in matches:
-        list_block = match[1]
-        list_items = re.findall(r"( {2}\* .+(?:\n {4}.+)*)", list_block)
+def _split_sections_by_heading(content: str) -> list[str]:
+    return re.split(r"(?=\n### )", content)
+
+
+def _extract_section_header(section: str) -> tuple[str, str] | None:
+    match = re.match(r"\n?### \*(\w+)\*\s+`([\w\.\:]+)`", section)
+    if match:
+        return (match.group(1), match.group(2))
+    match = re.match(r"\n?### `([\w\.\:]+)`", section)
+    if match:
+        return ("function", match.group(1))
+    return None
+
+
+def _format_default(obj: Any) -> str:
+    if isinstance(obj, (int, float, str, bool, type(None))):
+        return repr(obj)
+    if isinstance(obj, (list, dict, set, tuple)):
+        try:
+            return repr(obj)
+        except Exception:
+            pass
+    cls = obj.__class__
+    cls_md = f"{cls.__module__}.{cls.__qualname__}()"
+    return cls_md.replace("truss_chains.public_types", "truss_chains")
+
+
+def _get_model_field_defaults(full_class_path: str) -> dict[str, Any]:
+    try:
+        module_path, cls_name = full_class_path.rsplit(".", 1)
+        mod = importlib.import_module(module_path)
+        model_cls = getattr(mod, cls_name)
+        if issubclass(model_cls, pydantic.BaseModel):
+            defaults = {}
+            for name, field in model_cls.model_fields.items():
+                if field.default is not PydanticUndefined:
+                    value = field.default
+                elif field.default_factory is not None:
+                    value = field.default_factory()
+                else:
+                    continue
+
+                defaults[name] = _format_default(value)
+
+            return defaults
 
+        init_func = model_cls.__init__
+        sig = inspect.signature(init_func)
+        defaults = {}
+        for param_name, param in sig.parameters.items():
+            if param_name == "self":
```

Comment:
If `param.default == param.empty` is always true for `self`, do we need this check?